agents_infra/agents_infra/tests/unit/test_agent.py

```diff
@@ -73,11 +74,20 @@ def __del__(self):
         if hasattr(self, 'logfile'):
             os.remove(self.logfile.name)
 
+    def _ping_controller(self, url, api_key, timeout=3):
+        return super(MockAgent, self)._ping_controller(url, api_key, timeout)
```

Comment:
Consider whether this override is necessary. If no additional mocking behavior is introduced, this could be removed to avoid redundant pass-through code.
You could return `self.mock_controller_healthy` instead to simulate health status during unit tests.
```
def _ping_controller(self, url, api_key, timeout=3):
    return self.mock_controller_healthy
```

Reply:
👀